File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
def solution(queue1, queue2):
    length = len(queue1)
    total_length = length * 2
    s1, s2 = sum(queue1), sum(queue2)

    answer = 0
    q = queue1 + queue2
    p1 = 0
    for p2 in range(length, total_length):
        while s1 > s2:
            s1 -= q[p1]
            s2 += q[p1]
            p1 += 1
            answer += 1

        if s1 == s2:
            return answer

        s1 += q[p2]
        s2 -= q[p2]
        answer += 1

    return answer if answer >= total_length else -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int length = queue1.length;
        int totalLength = length * 2;
        
        Queue<Integer> q1 = new LinkedList<Integer>();
        Queue<Integer> q2 = new LinkedList<Integer>();
        
        long sum1 = 0, sum2 = 0;
        for (int i=0; i<length; i++) {
            q1.add(queue1[i]);
            q2.add(queue2[i]);
            
            sum1 += queue1[i];
            sum2 += queue2[i];
        }
        
        int answer = 0;
        int maxAnswer = totalLength * 2 + 1;
        while (answer < maxAnswer) {
            if (sum1 < sum2) {
                sum1 += q2.peek();
                sum2 -= q2.peek();
                
                q1.add(q2.poll());
                
                answer++;
            } else if (sum1 > sum2) {
                sum1 -= q1.peek();
                sum2 += q1.peek();
                
                q2.add(q1.poll());
                
                answer++;
            } else {
                return answer;
            }
        }
        
        return -1;
    }
}